graviti/dataset/version.py

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/env python3
+#
+# Copyright 2021 Graviti. Licensed under MIT License.
+#
+
+"""Related methods of the TensorBay version control."""
+
+from typing import Optional
+
+from tensorbay.client.lazy import PagingList
+from tensorbay.client.status import Status
+from tensorbay.client.struct import Branch, Commit, Draft, Tag
+
+
+class VersionControlMixin:  # pylint: disable=too-many-public-methods
+    """A mixin class supporting version control methods."""
+
+    _url: str
+    _access_key: str
+    _dataset_id: str
+    _status: Status
+
+    def checkout(self, revision: Optional[str] = None, draft_number: Optional[int] = None) -> None:
+        """Checkout to commit or draft.
+
+        Arguments:
+            revision: The information to locate the specific commit, which can be the commit id,
+                the branch, or the tag.
+            draft_number: The draft number.
+
+        """
+
+    def commit(self, title: str, description: str = "", *, tag: Optional[str] = None) -> None:
+        """Commit the draft.
+
+        Arguments:
+            title: The commit title.
+            description: The commit description.
+            tag: A tag for current commit.
+
+        """
+
+    def create_draft(
+        self, title: str, description: str = "", branch_name: Optional[str] = None
+    ) -> int:
+        """Create a draft.
+
+        Arguments:
+            title: The draft title.
+            description: The draft description.
+            branch_name: The branch name.
+
+        Return:
+            The draft number of the created draft.
+
+        """
+
+    def get_draft(self, draft_number: Optional[int] = None) -> Draft:
+        """Get the certain draft with the given draft number.
+
+        Arguments:
+            draft_number: The required draft number.
+                If is not given, get the current draft.
+
+        Return:
+            The :class:`.Draft` instance with the given number.
+
+        """
+
+    def list_drafts(
+        self, status: Optional[str] = "OPEN", branch_name: Optional[str] = None
+    ) -> PagingList[Draft]:
+        """List all the drafts.
+
+        Arguments:
+            status: The draft status which includes "OPEN", "CLOSED", "COMMITTED", "ALL" and None.
+                    where None means listing open drafts.
+            branch_name: The branch name.
+
+        Return:
+            The PagingList of :class:`drafts<.Draft>`.
+
+        """
+
+    def update_draft(
```

Comment:
will it be better we move the draft related operation into `Draft` instance?
```python
draft = dataset.get_draft()
draft.update()
draft.close()
...
```

Also consider commit branch and tag?